test/unit/rules/attributes-order-test.js

```diff
@@ -0,0 +1,785 @@
+              "severity": 2,
+              "source": "aria-label=\\"button\\"",
+            },
+            {
+              "column": 5,
+              "endColumn": 35,
+              "endLine": 1,
+              "filePath": "layout.hbs",
+              "isFixable": true,
+              "line": 1,
+              "message": "Modifier {{did-render this.someAction}} must go after attributes",
+              "rule": "attributes-order",
+              "severity": 2,
+              "source": "{{did-render this.someAction}}",
+            },
+          ]
+        `);
+      },
+    },
+    {
+      config: {
+        alphabetize: false,
+      },
+      template:
+        '<MyComponent @value="5" data-test-foo @change={{this.foo}} local-class="foo" {{on "click" this.foo}} ...attributes as |sth|>content</MyComponent>',
+      fixedTemplate:
+        '<MyComponent @value="5" @change={{this.foo}} data-test-foo local-class="foo" {{on "click" this.foo}} ...attributes as |sth|>content</MyComponent>',
+      verifyResults(results) {
+        expect(results).toMatchInlineSnapshot(`
+          [
+            {
+              "column": 38,
+              "endColumn": 58,
+              "endLine": 1,
+              "filePath": "layout.hbs",
+              "isFixable": true,
+              "line": 1,
+              "message": "Argument @change={{this.foo}} must go before attributes and modifiers",
+              "rule": "attributes-order",
+              "severity": 2,
+              "source": "@change={{this.foo}}",
+            },
+          ]
+        `);
+      },
+    },
+    {
+      template:
+        '<MyComponent @value="5" data-test-foo @change={{this.foo}} local-class="foo" {{on "click" this.foo}} as |sth|>content</MyComponent>',
+      fixedTemplate:
+        '<MyComponent @change={{this.foo}} @value="5" data-test-foo local-class="foo" {{on "click" this.foo}} as |sth|>content</MyComponent>',
+      verifyResults(results) {
+        expect(results).toMatchInlineSnapshot(`
+          [
+            {
+              "column": 38,
+              "endColumn": 58,
+              "endLine": 1,
+              "filePath": "layout.hbs",
+              "isFixable": true,
+              "line": 1,
+              "message": "Argument @change={{this.foo}} is not alphabetized",
+              "rule": "attributes-order",
+              "severity": 2,
+              "source": "@change={{this.foo}}",
+            },
+            {
+              "column": 38,
+              "endColumn": 58,
+              "endLine": 1,
+              "filePath": "layout.hbs",
+              "isFixable": true,
+              "line": 1,
+              "message": "Argument @change={{this.foo}} must go before attributes and modifiers",
+              "rule": "attributes-order",
+              "severity": 2,
+              "source": "@change={{this.foo}}",
+            },
+          ]
+        `);
+      },
+    },
+    {
+      config: {
+        attributeOrder: ['attributes'],
```

Comment:
What happens when not all types are ordered? Does it make sense?

Reply:
I've added a check to the configuration parsing to enforce that all types are included as it eliminates the need for guesswork.

Reply:
Nice. Can you add an `error` test case for this?